### ISSUE TITLE
Queue a task before resolving a promise

### ISSUE THREAD

**Issue by annevk** [immersive-web/webxr#1020]
You can no longer resolve a promise from "in parallel" directly. It requires a task that ideally identifies the task source and document it's associated with.

**Comment by Manishearth** [comment]
Makes sense. We don't have a task source we use right now: should we just make our own? Should we use the DOM task source for most things, and the user input one for input-related stuff.

**Comment by annevk** [comment]
Your own task source might make sense if in principle the tasks don't need to be ordered relative to other things. Gives user agents more freedom. (I suspect there might be more changes to this setup going forward, but this should at least set you up a bit better.)

**Comment by Manishearth** [comment]
Yeah there's very little dependent ordering

**Comment by Manishearth** [comment]
Filed a separate issue for the task source: https://github.com/immersive-web/webxr/issues/1022
